src/components/Error.tsx

```diff
@@ -0,0 +1,50 @@
+import { useContext, useEffect } from 'react';
+import { Close } from '@mui/icons-material';
+
+import { AppContext } from '../AppContext';
+
+type Error = {
+  type: string;
+  message: string;
+};
+
+export function Error() {
+  const { error, setError } = useContext(AppContext);
+
+  useEffect(() => {
+    // listen for errors from the service worker too
+    if ('serviceWorker' in navigator && navigator.serviceWorker.controller) {
+      navigator.serviceWorker.addEventListener('message', (event): void => {
+        if (
+          event.data &&
+          event.data.type === 'error' &&
+          'error' in event.data
```

Comment:
```suggestion
          event.data?.type === 'error' &&  event.data?.error
```

optional chaining syntax works quite nicely here